### PR TITLE
[Session replay] Support bg color + hidden nav bar

### PR DESCRIPTION
**Pull request by mariedm**
### What and why?

This PR adds support for hidden navigation bar and navigation bar with colors (both `backgroundColor` and `barTintColor`) for session replay.

See an example of session replay [here](https://mobile-integration.datadoghq.com/rum/replay/sessions/4218d75a-dbe8-451a-ab69-2549e9482cf6?applicationId=5e20092a-ba66-4868-bfa3-ee8a0f1a54a1&seed=003ccbcd-4599-4973-9d52-9a8b784111bf&ts=1716823072409) with different colors used.

Screenshot 2024-06-03 at 11 36 57

### How?

- If the navigation bar is hidden, it is identified as an `InvisibleElement` ([session replay example](https://mobile-integration.datadoghq.com/rum/replay/sessions/4218d75a-dbe8-451a-ab69-2549e9482cf6?applicationId=5e20092a-ba66-4868-bfa3-ee8a0f1a54a1&seed=003ccbcd-4599-4973-9d52-9a8b784111bf&ts=1716823072409))
- The recorder now accounts for the navigation bar's background color ([session replay example](https://mobile-integration.datadoghq.com/rum/replay/sessions/9956410b-35f8-41bd-8cba-8b66d234d50b?applicationId=5e20092a-ba66-4868-bfa3-ee8a0f1a54a1&seed=4fbb32e1-4da5-4aad-8f20-c1685d746e6d&ts=1716819952461))

### Context on Navigation Bar Style

The navigation bar can be styled (default or black, translucent or not) without specifying colors in the settings (background color, bar tint color). In these cases, the system assigns colors that are not reflected in the actual colors detected in code or view debugger—they remain nil. 

Since these colors are inaccessible, we must check the navigation bar settings directly, rather than using a general approach (eg. at the `UIViewRecorder` level).

### Note on Title's Font Color

Similarly, switching the nav style from Default to Black changes the title font color, but this change is not detected in the actual colors. Unless specified, the title’s font color is black with the Default style and white with the Black style. Since we have to check the navigation bar's style, we might have to add a property to `UINavigationBarWireframesBuilder`. The title's font color is not addresssed in this PR. 

### Note on Border Color

The border color is hardcoded to gray. We could use the background color, which usually fits the actual rendering better. However, with a white background, the gray border is necessary to distinguish the navigation bar's limit. I decided to leave it as is for now. Here are examples of session replay with [gray border](https://mobile-integration.datadoghq.com/rum/replay/sessions/1370124c-1e23-4967-abc8-2a32f1ca20a4?applicationId=5e20092a-ba66-4868-bfa3-ee8a0f1a54a1&seed=5dc4c194-e45c-4617-9580-28624d4d481d&ts=1716985728071) and [using the background color](https://mobile-integration.datadoghq.com/rum/replay/sessions/e6e15316-f190-40a7-ac07-6d725c18c416?applicationId=5e20092a-ba66-4868-bfa3-ee8a0f1a54a1&seed=1c8eb9c3-aaa0-4ab3-81d4-36ff8f36c6ed&ts=1716986561601).

### Tests

This PR adds new snapshot tests for testing the navigation bar background color. The `NavigationBars` storyboard adds a main view controller listing different navigation bar configurations in the same view. The other view controllers are embedded in a navigation controller to simulate a more realistic use case.

image

image

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests for Core, RUM, Trace, Logs, CR and WVT
- [ ] Run unit tests for Session Replay
- [ ] Run integration tests
- [ ] Run smoke tests
- [ ] Run tests for `tools/`

### CODE REVIEW

**Comment by maciejburda** [comment]
It's failiing on `testNavigationBars` I think we need to commit the snapshot hash

File: DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests/Utils/SnapshotTestCase.swift [hunk 1]
```diff
@@ -35,6 +35,21 @@ internal class SnapshotTestCase: XCTestCase {
         return viewController
     }
 
+    func showNavigationController(for fixture: Fixture) -> UINavigationController? {
```
Comment (maxep) [inline DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests/Utils/SnapshotTestCase.swift:38]: /question This method and `AppDelegate.showNavigationController` are not used, should they be removed?

Comment (mariedm) [reply DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests/Utils/SnapshotTestCase.swift]: Totally, I forgot to delete them. It's fixed now ✅

**Review by maxep — APPROVED** [review]
Looks great, nice work on the snapshot tests!!

**Review by maciejburda — APPROVED** [review]
Nice one! 💪

**Pull request merged**